build-system/tasks/build.js

```diff
@@ -67,50 +88,43 @@ async function performBuild(watch, defaultTask) {
   process.env.NODE_ENV = 'development';
   printNobuildHelp();
   printConfigHelp(defaultTask ? 'gulp' : watch ? 'gulp watch' : 'gulp build');
-  parseExtensionFlags(defaultTask);
-  return Promise.all([compileCss(watch), compileJison()])
-    .then(() => {
-      return Promise.all([
-        polyfillsForTests(),
-        buildAlp({watch}),
-        buildExaminer({watch}),
-        buildWebWorker({watch}),
-        defaultTask ? Promise.resolve() : buildExtensions({watch}),
-        compileAllUnminifiedTargets(watch),
-      ]);
-    })
-    .then(() => {
-      if (isTravisBuild()) {
-        // New line after all the compilation progress dots on Travis.
-        console.log('\n');
-      }
-    });
-}
-
-/**
- * Compile the polyfills script and drop it in the build folder
- * @return {!Promise}
- */
-function polyfillsForTests() {
-  return compileJs('./src/', 'polyfills.js', './build/');
+  if (defaultTask) {
+    printDefaultTaskHelp();
+  }
+  if (!argv.lazy_build) {
+    parseExtensionFlags();
+  }
+  await Promise.all([compileCss(watch), compileJison()]);
+  await Promise.all([
+    bootstrapThirdPartyFrames(watch),
+    compileCoreRuntime(watch),
+  ]);
+  if (!defaultTask) {
+    await compileUnminifiedRuntimeTargets(watch);
+    await buildExtensions({watch});
+  }
+  if (isTravisBuild()) {
+    // New line after all the compilation progress dots on Travis.
+    console.log('\n');
+  }
 }
 
 /**
  * The default task run when `gulp` is executed
  */
 async function defaultTask() {
   await watch(/* defaultTask */ true);
-  serve(argv.lazy_build_extensions);
-  const startedMessage = green('Started ') + cyan('gulp ') + green('server. ');
-  if (argv.lazy_build_extensions) {
+  serve(argv.lazy_build);
+  log(green('Started ') + cyan('gulp ') + green('server. '));
+  if (argv.lazy_build) {
     log(
-      startedMessage +
-        green(
-          'Extensions will be lazily built when requested from the server...'
-        )
+      green(
+        'Runtime targets and extensions will be lazily built when requested...'
```

Comment:
Maybe just "JS files". "Runtime targets" is jargon

Reply:
Done.